# CustomerDemographic Sheet

In [20]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_excel("KPMG_VI_New_raw_data_update_final.xlsx", sheet_name= "CustomerDemographic",skiprows=1)
df.head(10)

customer_id      first_name  last_name  gender  \
0            1         Laraine  Medendorp       F   
1            2             Eli    Bockman    Male   
2            3           Arlin     Dearle    Male   
3            4          Talbot        NaN    Male   
4            5  Sheila-kathryn     Calton  Female   
5            6            Curr  Duckhouse    Male   
6            7            Fina     Merali  Female   
7            8             Rod      Inder    Male   
8            9            Mala       Lind  Female   
9           10        Fiorenze    Birdall  Female   

   past_3_years_bike_related_purchases        DOB  \
0                                   93 1953-10-12   
1                                   81 1980-12-16   
2                                   61 1954-01-20   
3                                   33 1961-10-03   
4                                   56 1977-05-13   
5                                   35 1966-09-16   
6                                    6 1976-02-23   
7                                   31 1962-03-30   
8                                   97 1973-03-10   
9                                   49 1988-10-11   

                              job_title job_industry_category  \
0                   Executive Secretary                Health   
1                Administrative Officer    Financial Services   
2                    Recruiting Manager              Property   
3                                   NaN                    IT   
4                         Senior Editor                   NaN   
5                                   NaN                Retail   
6                                   NaN    Financial Services   
7                       Media Manager I                   NaN   
8  Business Systems Development Analyst           Argiculture   
9               Senior Quality Engineer    Financial Services   

      wealth_segment deceased_indicator  \
0      Mass Customer                  N   
1      Mass Customer                  N   
2      Mass Customer                  N   
3      Mass Customer                  N   
4  Affluent Customer                  N   
5     High Net Worth                  N   
6  Affluent Customer                  N   
7      Mass Customer                  N   
8  Affluent Customer                  N   
9      Mass Customer                  N   

                                             default owns_car  tenure  
0                                                 "'      Yes    11.0  
1                       <script>alert('hi')</script>      Yes    16.0  
2                                2018-02-01 00:00:00      Yes    15.0  
3  () { _; } >_[$($())] { touch /tmp/blns.shellsh...       No     7.0  
4                                                NIL      Yes     8.0  
5                                           ðµ ð ð ð      Yes    13.0  
6                                          â°â´âµâââ      Yes    11.0  
7                            (â¯Â°â¡Â°ï¼â¯ï¸µ â»ââ»)       No     7.0  
8                                                0/0      Yes     8.0  
9                                               ð©ð½      Yes    20.0

In [35]:
df.shape

(4000, 13)

### Checking Missing Values

In [13]:
missing_values = df.isnull().sum()
missing_values[missing_values>0]

last_name                125
DOB                       87
job_title                506
job_industry_category    656
default                  302
tenure                    87
dtype: int64

Following Column has missing values:
- last_name
- DOB
- job_title
- job_industry_category 
- default
- tenure 

All these Data fields must be completed to be able to drive a better insight from the data. (Completeness)

### Gender Column

In [24]:
df['gender'].unique()

array(['F', 'Male', 'Female', 'U', 'Femal', 'M'], dtype=object)

Here we can see three issues : 
- The values are not consistent, gender is either have to label Male/Female or M/F not both.
- The inaccuracy/incorrect data in the datafield, gender column is labelled as Femal
- Irrelevant value in the dataset which is value U in gender column.

### DOB Column

In [28]:
df[['DOB']].head() # quick look at the dataset

DOB
0 1953-10-12
1 1980-12-16
2 1954-01-20
3 1961-10-03
4 1977-05-13

In [29]:
df['DOB'].dtype #checking datatype

dtype('<M8[ns]')

In [32]:
year = pd.DatetimeIndex(df['DOB']).year #extracting year
year.sort_values().to_list()[0:10]

[1843.0,
 1931.0,
 1935.0,
 1940.0,
 1943.0,
 1944.0,
 1953.0,
 1953.0,
 1953.0,
 1953.0]

We can see that one of the year is labeled as 1843 it means they are 175 years old which is obviously not correct.

### Job_title Column

In [33]:
df[['job_title']].head(20)

job_title
0                    Executive Secretary
1                 Administrative Officer
2                     Recruiting Manager
3                                    NaN
4                          Senior Editor
5                                    NaN
6                                    NaN
7                        Media Manager I
8   Business Systems Development Analyst
9                Senior Quality Engineer
10                                   NaN
11                Nuclear Power Engineer
12                           Developer I
13                     Account Executive
14                      Junior Executive
15                      Media Manager IV
16                       Sales Associate
17                             Professor
18                   Geological Engineer
19                       Project Manager

Here we can see many job_title data has number in their job role which is not correct.

### Default Column

In [34]:
df[['default']].head()

default
0                                                 "'
1                       <script>alert('hi')</script>
2                                2018-02-01 00:00:00
3  () { _; } >_[$($())] { touch /tmp/blns.shellsh...
4                                                NIL

Here we can see the non-validity of the data field because it has no meaningful data in it.